### ISSUE TITLE
SPARQL: The unnamed/default graph should have a standard name

### ISSUE THREAD

**Issue by dbooth-boston** [w3c/EasierRDF#44]
At present the unnamed/default graph has no standard name.  This means that, when writing code that manipulates graphs, one must special-case the unnamed/default graph.  It also violates one of the [Axioms of Web Architecture: "Any resource of significance should be given a URI."](https://www.w3.org/DesignIssues/Axioms.html#Universality)

I think the unnamed/default graph should have a standard name, such as http://www.w3.org/1999/02/22-rdf-syntax-ns#defaultGraph ( rdf:defaultGraph ).  Implied references to the unnamed/default graph in SPARQL, TriG, etc., should be understood as short-hand for this graph name.

**Comment by kasei** [comment]
Does this imply that you think disparate endpoints would be explicitly using the *same* default graph? That sounds rather strange to me.

One of the underlying setups we attempted to support in the SPARQL 1.1 WG was systems which had an underlying quadstore (where every graph has a name) in which the query engine would use one specific named graph as the default. I've designed systems like this in the past, and been very happy with it. I think an alternative approach to this issue (defined by a future version of SPARQL) might be to reconsider the default graph as a pre-defining of the active graph (as if the query were wrapped in a `GRAPH <g> { ... }` block). The specific graph being used could be indicated in the service description, allowing it to be referenced explicitly.

**Comment by dbooth-boston** [comment]
> Does this imply that you think disparate endpoints would be explicitly using the same default graph?

It would be the same as when different SPARQL endpoints use <urn:foo> as a named graph name.  A query would not magically cause all SPARQL endpoints to return results from all SPARQL endpoints that use that graph name.

**Comment by kasei** [comment]
No, but it might cause some surprising results if you start dealing with metadata or provenance data about graphs (where statements made about another endpoint's default graph now apply to all default graphs everywhere).

**Comment by kasei** [comment]
Alternatively, for endpoints that are using the SPARQL Protocol, a default graph IRI could be constructed based on the service endpoint URL.

**Comment by rnavarropiris** [comment]
> As the Dedicated Unnamed Default Graph is not referenceable, it is not possible to join it with other graphs. In other words, if a query specifies any other graph (whether using FROM or FROM NAMED) the default dataset of the service will be overwritten (13.2 Specifying RDF Datasets) and therefore the Dedicated Unnamed Default Graph is not accessible anymore.

by @depressiveRobot, article [here](http://depressiverobot.com/2015/07/29/sparql-datasets.html)

I complete agree with @dbooth-boston, the default graph should be referenceable, so that it could also be used in a dataset definition.

Furthermore, the query dataset could then be defined as the [union default graph](https://www.w3.org/TR/2013/REC-sparql11-service-description-20130321/#sd-uniondefaultgraph), since otherwise there would be no way to retrieve the list of existing graphs in the quad store with a query e.g. in the form 

```
SELECT DISTINCT ?g { GRAPH ?g {?s ?p ?o}}
```
